RequestService/RequestService.Repo/Repository.cs

```diff
@@ -997,5 +1057,45 @@ public async Task<int> GetRequestIDFromGuid(Guid guid)
                 return -1;
             }
         }
+
+        public List<ShiftJob> GetUserShiftJobsByFilter(GetUserShiftJobsByFilterRequest request)
+        {
+            byte requestTypeShift = (byte)RequestType.Shift;
+            var jobs = _context.Job
+                .Include(i => i.NewRequest)
+                .ThenInclude(i => i.Shift)
+                .Where(x => x.VolunteerUserId == request.VolunteerUserId && x.NewRequest.RequestType == requestTypeShift).ToList();
+
+            if (jobs == null || jobs.Count == 0)
+            {
+                return null;
+            }
+
+            if (request.JobStatusRequest.JobStatuses.Count > 0)
+            {
+                jobs = jobs.Where(x => request.JobStatusRequest.JobStatuses.Contains((JobStatuses)x.JobStatusId)).ToList();
+            };
+
+            if (request.DateFrom.HasValue)
+            {
+                jobs = jobs.Where(x => x.NewRequest.Shift.StartDate >= request.DateFrom.Value).ToList();
```

Comment:
I think this line needs end date and line 1086 needs start date, to be more inclusive.